app/javascript/controllers/shortened_link_form_controller.js

```diff
@@ -0,0 +1,50 @@
+import { Controller } from "stimulus"
+import StimulusReflex from 'stimulus_reflex'
+
+export default class extends Controller {
+  static targets = ['name', 'shortenedPath', 'originalUrl']
+
+  connect () {
+    StimulusReflex.register(this)
+  }
+
+  closeModal() {
+    this.stimulate('ShortenedLinkFormReflex#close_modal')
+  }
+
+  trapClick(e) {
+    e.stopPropagation()
+  }
+
+  handleKeyDown(e) {
+    if (e.key === 'Escape') {
+      this.closeModal()
+    }
+  }
+
+  validate(e) {
+    this.stimulate('ShortenedLinkFormReflex#validate', [e.target], this.shortenedLinkParams)
+  }
+
+  save(e) {
+    this.stimulate('ShortenedLinkFormReflex#save', [e.target], this.shortenedLinkParams)
+  }
+
+  destroy
+
+  get shortenedLinkId() {
+    const id = parseInt(this.data.get('linkId'))
```

Comment:
```suggestion
    const id = parseInt(this.data.get('linkId'), 10)
```

Reply:
Isn't 10 the default value?

Reply:
it's actually derived from the string passed in the first param

Reply:
Never trust Javascript